### PR TITLE
Replace custom useForm with react-hook-form, add title validation for topics

### PR DESCRIPTION
**Pull request by antoniosaleme**
## Description

This Pull Request implements the following changes:

* Replaces the custom useForm hook with the [react-hook-form](https://react-hook-form.com/)  library for form management and validation. This provides a more robust and standardized solution for form handling, making the codebase more maintainable and easier to understand.
* Adds a validation check to the "Create a Topic" form to ensure that the topic title is not empty. If the user tries to submit a topic with an empty title, an error message is displayed, and the form submission is prevented.

Please note that no styling updates have been made in this PR. The addition of appropriate styles for the form and error messages should be addressed in a subsequent PR to enhance the user experience further.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots (if applicable)

Screenshot 2023-04-16 at 14 41 57

Screenshot 2023-04-16 at 14 41 11

## How has this PR been Tested?

- [x] Manually

### CODE REVIEW

File: client/src/utils/validations.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const isValidEmail = (email) => {
+  const match = String(email)
+    .toLowerCase()
+    .match(
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+    );
+
+  return !!match;
```
Comment (Pablo-Mdz) [inline client/src/utils/validations.js]: why !!match ? what that means  !! ?

File: client/src/components/Topic/CreateATopic.jsx [hunk 2]
```diff
@@ -3,9 +3,14 @@ import axios from 'axios';
 
 export default function CreateATopic({ onTopicCreated, hideCreateTopic }) {
   const [title, setTitle] = useState('');
+  const [error, setError] = useState('');
 
   const handleSubmit = (event) => {
     event.preventDefault();
+    if (title.trim() === '') {
```
Comment (Pablo-Mdz) [inline client/src/components/Topic/CreateATopic.jsx:10]: i like this way

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 3]
```diff
@@ -2,24 +2,23 @@ import { useState, useContext } from 'react';
...
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
 
   const { storeToken, verifyStoredToken } = useContext(AuthContext);
 
   const navigate = useNavigate();
 
-  const handleLoginSubmit = (e) => {
-    e.preventDefault();
```
Comment (Pablo-Mdz) [inline client/src/pages/LoginPage/LoginPage.jsx:20]: why did you delete this e.preventDefault() ? is not necessary on this function?

File: client/src/pages/LoginPage/LoginPage.jsx [hunk 4]
```diff
@@ -44,43 +42,58 @@ function LoginPage() {
               Sign in to your account
             </h2>
           </div>
-          <form onSubmit={handleLoginSubmit} className="mt-8 space-y-6">
+          <form
+            onSubmit={handleSubmit(handleLoginSubmit)}
```
Comment (Pablo-Mdz) [inline client/src/pages/LoginPage/LoginPage.jsx:49]: is usefull to do on this way? is not more clean and easy use only "required"?

**Review by Pablo-Mdz — APPROVED** [review]

File: client/src/pages/LoginPage/LoginPage.jsx (hunk 3 above)
Comment (antoniosaleme) [reply client/src/pages/LoginPage/LoginPage.jsx:20]: If I'm not wrong react-hook-form is doing this job

File: client/src/pages/LoginPage/LoginPage.jsx (hunk 4 above)
Comment (antoniosaleme) [reply client/src/pages/LoginPage/LoginPage.jsx:49]: is the way react-hook-form works

File: client/src/utils/validations.js (hunk 1 above)
Comment (antoniosaleme) [reply client/src/utils/validations.js]: I don't longer this need this file, I will delete it
